resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
+            ],
+            "Resource": "arn:aws:ec2:*:*:prefix-list/*"
+        },
+        {
+            "Sid": "Image",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImages"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ImagaeAttibute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImageAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:image/*"
+        },
+        {
+            "Sid": "IdFormat",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribePrincipalIdFormat",
+                "ec2:DescribeIdentityIdFormat",
+                "ec2:DescribeIdFormat",
+                "ec2:DescribeAggregateIdFormat"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "AddressAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAddressesAttribute"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Addresses",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAddresses"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "AccountAttributes",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeAccountAttributes"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ByoipCidrs",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeByoipCidrs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CapacityReservation",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCapacityReservations"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CarrierGateways",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCarrierGateways"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Instances",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeClassicLinkInstances"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CustomerGateways",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCustomerGateways"
+            ],
             "Resource": "*"
         },
         {
+            "Sid": "InstanceTypeOfferings",
             "Effect": "Allow",
             "Action": [
-                "s3:ListBucket"
+                "ec2:DescribeInstanceTypeOfferings"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "KeyPairs",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeKeyPairs"
+            ],
+            "Resource": "*"
```

Comment:
Any way this can be more restricted?

Reply:
cross-checking with https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html#amazonec2-policy-keys it doesn't seem like it can be more restrictive